beamer/state_machine.py

```diff
@@ -262,8 +265,13 @@ def _handle_request_resolved(event: RequestResolved, context: Context) -> Handle
 def _handle_hash_invalidated(event: HashInvalidated, context: Context) -> HandlerResult:
     claim = _find_claim_by_fill_hash(context, event.fill_hash)
     if claim is not None:
-        claim.invalidate()
-        return True, None
+        request = context.requests.get(claim.request_id)
+        assert request is not None, "Request missing for claim"
+
+        fill_hash_of_claim = request.fill_hash_with_fill_id(claim.latest_claim_made.fill_id)
+        if fill_hash_of_claim == event.fill_hash and claim.is_started:
```

Comment:
this is already covered in `_find_claim_by_fill_hash`

Reply:
Thanks, overlooked this.